rust/api.md

```diff
@@ -0,0 +1,532 @@
+            <a href="#password">"password"</a>: string,
+            <a href="#private">"private"</a>: string
+          },
+          <a href="#snmp">"snmp"</a>: {
+            <a href="#username">"username"</a>: string,
+            <a href="#password">"password"</a>: string,
+            <a href="#community">"community"</a>: string,
+            <a href="#auth_algorithm">"auth_algorithm"</a>: string,
+            <a href="#privacy_password">"privacy_password"</a>: string,
+            <a href="#privacy_algorithm">"privacy_algorithm"</a>: string
+          }
+        }]
+      }],
+      <a href="#excluded">"excluded"</a>: [string],
+      <a href="#scanner_parameters">"scanner_parameters"</a>: {key:value},
+      <a href="#vts">"vts"</a>: {
+        <a href="#vt_single">"vt_single"</a>: [{
+          <a href="#oid">"oid"</a>: string,
+          <a href="#vt_parameters">"vt_parameters"</a>: [{
+            <a href="#parameter">"parameter"</a>: string,
+            <a href="#value">"value"</a>: any
+          }]
+        }],
+        <a href="#vt_group">"vt_group"</a>: [string]
+      }
+    }
+  </code>
+</pre>
+- Status codes: 
+  - 200 Ok
+  - 400 Bad Request
+- Response: None
+
+
+### get scan
+
+The get scan command returns information about a given scan. By default only basic information about the scan are requested. With the fields details the command also returns results.
+
+- Path: `/scan/get`
+- Request:
+<pre>
+  <code>
+    {
+      <a href="#scan_id">"scan_id"</a>: string,
+      <a href="#get_results">"get_results"</a>: bool,
+      <a href="#pop_results">"pop_results"</a>: bool,
+      <a href="#get_host_info">"pop_results"</a>: bool
+    }
+  </code>
+</pre>
+- Status codes:
+  - 200 Ok
+  - 400 Bad Request
+  - 404 Scan not found
+- Response:
+<pre>
+  <code>
+    {
+      <a href="#scan_id">"scan_id"</a>: string,
+      <a href="#start_time">"start_time"</a>: int,
+      <a href="#end_time">"end_time"</a>: int,
+      <a href="#status">"status"</a>: string,
+      <a href="#progress">"progress"</a>: int,
+      <a href="#results">"details"</a>: [{
+        <a href="#ip_address">"ip"</a>: string,
+        <a href="#host">"host"</a>: string,
+        <a href="#port">"port"</a>: int,
+        <a href="#protocol">"protocol"</a>: string,
+        <a href="#type">"type"</a>: string,
+        <a href="#name">"name"</a>: string,
+        <a href="#description">"description"</a>: string
+      }],
+      <a href="#host_info">"host_info"</a>: {
+        <a href="#alive_hosts">"alive_hosts"</a>: int,
+        <a href="#dead_hosts">"dead_hosts"</a>: int,
+        <a href="#excluded_hosts">"excluded_hosts"</a>: int,
+        <a href="#total_hosts">"total_hosts"</a>: int
+      }
+    }
+  </code>
+</pre>
+
+
+### delete scan
+
+The delete scan request deletes a scan from the scanner including all data corresponds to it. Unless the scan is not finished, this request will fail.
+
+- Path: `/scan/delete`
```

Comment:
Here I would also use a path parameter and no action duplication instead of json:
```
DELETE /scan/{id}
```